app/models/jwt_secret.rb

```diff
@@ -0,0 +1,21 @@
+class JwtSecret < ApplicationRecord
+  include Encryptable
+
+  encrypts :token
+
+  belongs_to :user, inverse_of: :jwt_secret
+
+  validates :token, uniqueness: true
+  validates :user, presence: true
+
```

Comment:
is there a notion of token expiry based on time? I wonder if we can use jwt with the new login page, but not sure how to maintain session expiry

Reply:
Yeah, jwt tokens support expiry. We can add that later. This PR just allows users to sign in with tokens but tokens aren't issued anywhere right now. That's something we need to add in a follow up PR.

Reply:
I am working on SSO integration with foreman via keycloak, I will take care of the token expiry and related things :)

Reply:
Excellent, as a quick hint: This should do the trick:

https://github.com/timogoebel/foreman_register/blob/d7acbed10f9ec9c85891ec8823bf01b70f607d8a/app/services/foreman_register/registration_token.rb#L13-L26